libipl/libipl.H

```diff
@@ -18,11 +21,50 @@ enum ipl_type {
 #define IPL_INFO    1
 #define IPL_DEBUG   2
 
+//IPL Error types
+enum ipl_error_type {
+	IPL_NO_ERR = 0,
+	IPL_HWP_ERR,
+	IPL_SBE_CHIPOP_ERR,
+	IPL_SBE_BOOT_FAIL,
+	IPL_PRI_PROC_NON_FUNC,
+	IPL_FSI_TGT_NOT_FOUND,
+	IPL_CFAM_ERR,
+};
+
+typedef std::map<ipl_error_type, const char*> err_msg_map_type;
+const err_msg_map_type err_msg_map = {
+	{IPL_NO_ERR, ""},
+	{IPL_HWP_ERR, "HWP execution reported error"},
+	{IPL_SBE_CHIPOP_ERR, "SBE chip-op reported error"},
+	{IPL_SBE_BOOT_FAIL, "SBE Boot failure"},
+	{IPL_PRI_PROC_NON_FUNC, "Primary processor is non functional"},
+	{IPL_FSI_TGT_NOT_FOUND, "FSI target is not available"},
+	{IPL_CFAM_ERR, "CFAM read/write failure"},
+};
```

Comment:
I really don't like to see code in header file.  C++ has messed up that completely.

Reply:
This need to be shared with user's. that is why i placed here.